pkgs/applications/misc/phoc/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, meson
+, ninja
+, pkg-config
+, python3
+, wrapGAppsHook
+, libinput
+, gnome3
+, glib
+, gtk3
+, wayland
+, dbus
+, cmake
+, libdrm
+, libxkbcommon
+, wlroots
+}:
+
+let
+  # See upstream report and fix in Alpine
+  # - https://source.puri.sm/Librem5/phosh/-/issues/422
+  # - https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/14522
+  phocWlroots = wlroots.overrideAttrs (old: {
+    patches = (old.patches or []) ++ [
+      ./0001-Revert-layer-shell-error-on-0-dimension-without-anch.patch
+      ./0002-stop-including-eglmesaext-h.patch
```

Comment:
Where does this patch come from?

Reply:
The first one is from the MR linked in the comment, and the second one is https://github.com/swaywm/wlroots/commit/e18599b05e0f0cbeba11adbd489e801285470eab to resolve the missing `EGL/eglmesaext.h` build error (`WLR_HAS_EGLMESAEXT_H` is somehow set in the build process).

Reply:
Then we should include that in a comment on the line before the patch. Also maybe we could just use fetchpatch.

Reply:
Sure, updated PR to use `fetchpatch`.